sdk/Trace/MultiSpanProcessor.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk\Trace;
+
+use OpenTelemetry\Trace as API;
+
+/**
+ * Class MultiSpanProcessor is a SpanProcessor that forwards all events to an
+ * array of SpanProcessors.
+ */
+final class MultiSpanProcessor implements SpanProcessor
```

Comment:
Do you think SpanProcessorsCollection will be a better name?

Reply:
Is there a reason you're interested in changing this @beniamin?

Reply:
MultiSpanProcessor is taking me to the idea that this is a processor that processes multiple spans. In reality this class is a collection of processors.

Reply:
Sounds like a decent compromise to me; you have thoughts @morrisonlevi?

Reply:
I tend to agree, the `MultiSpanProcessor` name is a bit misleading. Also, in this design the `MultiSpanProcessor` isn't **actually** a `SpanProcessor` at all. Although it fits the API/interface of a `SpanProcessor` it is really just a meta-object used by the `TracerProvider` to delegate the calling of `onStart` `onEnd` and `shutdown` to the actual `SpanProcessors`.

It makes sense to delegate this behavior outside of the `TracerProvider` proper and let a meta object handle the actual work of dispatching calls to all the registered `SpanProcessors`. I scratched my head for too long to think of a name that matched this delegated fan-out logic but they were all clunkier than `SpanProcessorsCollection`

Reply:
SpanMultiProcessor?

Reply:
I don't *love* it ... but that's alright, something about the use of `Multi` that throws my brain off. It still works though. Honestly, naming is hard and I don't want to beleaguer the point too much when I don't think the end result will be substantially better. `SpanMultiProcessor` works for me @bobstrecansky @beniamin thoughts on `SpanMultiProcessor`?

Reply:
That's totally fine with me.

Reply:
I’m ok with it.